pkg/kubeapiserver/admission/config.go

```diff
@@ -47,7 +47,7 @@ type Config struct {
 }
 
 // New sets up the plugins and admission start hooks needed for admission
-func (c *Config) New(proxyTransport *http.Transport, egressSelector *egressselector.EgressSelector, serviceResolver webhook.ServiceResolver, tp *trace.TracerProvider) ([]admission.PluginInitializer, genericapiserver.PostStartHookFunc, error) {
+func (c *Config) New(proxyTransport *http.Transport, egressSelector *egressselector.EgressSelector, serviceResolver webhook.ServiceResolver, tp trace.TracerProvider) ([]admission.PluginInitializer, genericapiserver.PostStartHookFunc, error) {
```

Comment:
Just a small quick question. Why doesn't the input trace.TracerProvider parameters here use pointer types? I am working on a similar feature. I happened to see you modify here. So wanted to ask your idea.

Reply:
`trace.TracerProvider` [is an interface](https://pkg.go.dev/go.opentelemetry.io/otel/trace#TracerProvider).  I think it would typically be most appropriate to receive the interface and not a pointer to the interface.

Reply:
*I think this is the answer but anyone can correct me here or make it clearer, please*
During reviews it was decided to modify from the pointer parameter, as `pointers to interfaces` are very rarely required and are discouraged. ([Here's a good explanation I learned from](https://go.dev/doc/faq#pointer_to_interface)) Instead, methods that satisfy interfaces usually have pointer receivers. This is because if the method needs to modify the receiver, the receiver must be a pointer, because changes to that receiver will be invisible to the caller. Here's another good post about how [interfaces are pointers themselves](https://www.timr.co/go-interfaces-the-tricky-parts/#:~:text=Go's%20interface%20values%20are%20really,type%20of%20the%20concrete%20value.) (and a pointer to an interface would be a double pointer).

Here's the [TracerProvider interface](https://github.com/kubernetes/kubernetes/blob/d0c92aedbab1bd448a5271496689ae2a3f11debd/vendor/go.opentelemetry.io/otel/trace/trace.go#L663) - the parameter highlighted above
Here's the [TracerProvider struct](https://github.com/kubernetes/kubernetes/blob/eefcf6aa801c5db48b564d0464470d623b8bfb79/vendor/go.opentelemetry.io/otel/sdk/trace/provider.go#L109)
You can see the go sdk also does not use pointer to the TracerProvider as param, like here [WithTracerProvider](https://github.com/kubernetes/kubernetes/blob/d0c92aedbab1bd448a5271496689ae2a3f11debd/vendor/go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp/config.go#L86)

Reply:
@sallyom Thanks sallyom. The reason you gave makes so much sense! This pr is perfect.